jupyterlab_caip_optimizer/src/service/optimizer.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable @typescript-eslint/camelcase */
+import { ServerConnection } from '@jupyterlab/services';
+
+import { TransportService, handleApiError } from 'gcp_jupyterlab_shared';
+import { Study, MetadataFull, MetadataRequired } from '../types';
+
+// const AI_PLATFORM = 'https://ml.googleapis.com/v1';
+
+function zoneToRegion(zone: string): string {
+  const divider = '-';
+  const region = zone
+    .split(divider)
+    .slice(0, 2)
+    .join(divider);
+  return region;
+}
+
+/**
+ * Class to interact with Optimizer
+ */
+export class OptimizerService {
+  private readonly serverSettings = ServerConnection.defaultSettings;
+
+  constructor(private _transportService: TransportService) {}
+
+  /** TODO: Remove hardcoded values and 'option' parameter before release. Currently implemented this way because of ssh restrictions to GCP */
+  async getMetaData(option: string = 'default-metadata'): Promise<MetadataRequired> {
+    try {
+      if (option === 'default-metadata') {
+        const metadata_default: MetadataRequired = {
```

Comment:
I'm seeing some irregularities in case styles (e.g. snake case here, camel case in other files), I think it would be worth it to figure out a consistent case style for this project.

Also somewhat related, figure out when to make consts all caps (like the endpoints const in this file). I believe the convention for our internal source code is that only global constants are capitalized, but definitely feel free to define your own rules

Reply:
Sounds good. I've tried following those (e.g. capitalizing enums) but I'll try to be more consistent.